src/logoutd.c

```diff
@@ -54,8 +54,7 @@ static int check_login (const struct utmp *ut)
 	/*
 	 * ut_user may not have the terminating NUL.
 	 */
-	strncpy (user, ut->ut_user, sizeof (ut->ut_user));
-	user[sizeof (ut->ut_user)] = '\0';
+	shdw_strscpy (user, ut->ut_user, sizeof (ut->ut_user) + 1);
```

Comment:
`STRSCPY (user, ut->ut_user)`?

Reply:
So, the array is of size `UT_NAMESIZE`, which in my sistem happens to be 32.  `utmp(7)` says that the string is terminated iff the name is shorter than 32 chars.  If the name is 32 chars, we're out of luck.  I think it was a bad decission to not include the null byte inconditionally, and thus limiting the name to 31 bytes.

This made me realize that `strscpy()` seems to not be useful for this case: it's introducing a bug.

When the name is 32 chars, we were passing 33 to `strscpy()`, but that might crash, or would copy 33 chars from the source string and then zero the last one.

For a correct function here, we would need a new function which accepts 2 different sizes, one for the dest buffer, and one for the source buffer.